Social-Music/View Controllers/ProfileViewController.m

```diff
@@ -66,19 +68,47 @@ - (void)viewDidLoad {
 - (void)viewDidAppear:(BOOL)animated {
     self.accessToken = [[SpotifyManager shared] accessToken];
     NSLog(@"access token: %@", self.accessToken);
-
-    [[TopItems shared] fetchTopData:@"artists" completion:^{
-        self.currUserArtistData = [[TopItems shared] artistData];
-        NSLog(@"%@", self.currUserArtistData);
-    }];
-    
-    [[TopItems shared] fetchTopData:@"tracks" completion:^{
-        self.currUserTrackData = [[TopItems shared] trackData];
-        NSLog(@"%@", self.currUserTrackData);
-    }];
     
-    [self.favoritesTableView reloadData];
+    PFUser *curr = PFUser.currentUser;
+
+    if (curr[@"topArtists"] == nil && curr[@"topSongs"] == nil) {
+        [[TopItems shared] fetchTopData:@"artists" completion:^{
+            self.currUserArtistData = [[TopItems shared] artistData];
+            self.currUserArtistPhotos = [[TopItems shared] artistPhotos];
+            NSLog(@"profile artist data: %@", self.currUserArtistData);
+        }];
+        
+        [[TopItems shared] fetchTopData:@"tracks" completion:^{
+            self.currUserTrackData = [[TopItems shared] trackData];
+            self.currUserTrackPhotos = [[TopItems shared] trackPhotos];
```

Comment:
Is there a reason you are persisting this data in a singleton, `TopItems`? Instead of just fetching the data and storing within this view controller?

Reply:
I moved top items to its own file just because the profile view controller code was becoming very lengthy - I can move it back if it makes more sense to have the code in the vc.

Reply:
Having the code separate is good, my question was more regarding why it was a singleton, and why you are saving the data of the requests in the singleton.

Reply:
At first, I was thinking that would make the top data easy to access from any of the vc's like the matches details vc. I ended up not needing to use it from vc's other than the profile vc since I pulled from the database for matches details vc. The reason I'm saving the data of the requests is so I can pass the photo data and top items data over to the profile vc but I think the models we were talking about implementing would take care of that in a better way.